ports/stm32/boards/ARDUINO_GIGA/mpconfigboard.h

```diff
@@ -0,0 +1,317 @@
+
+#define MICROPY_HW_SDRAM_AUTOREFRESH_NUM    (8)
+#define MICROPY_HW_SDRAM_BURST_LENGTH       1
+#define MICROPY_HW_SDRAM_REFRESH_RATE       (64) // ms
+
+// SDRAM configuration
+#define MICROPY_HW_FMC_SDCKE0       (pin_H2)
+#define MICROPY_HW_FMC_SDNE0        (pin_H3)
+#define MICROPY_HW_FMC_SDNBL0       (pin_E0)
+#define MICROPY_HW_FMC_SDNBL1       (pin_E1)
+#define MICROPY_HW_FMC_SDCLK        (pin_G8)
+#define MICROPY_HW_FMC_SDNCAS       (pin_G15)
+#define MICROPY_HW_FMC_SDNRAS       (pin_F11)
+#define MICROPY_HW_FMC_SDNWE        (pin_H5)
+#define MICROPY_HW_FMC_BA0          (pin_G4)
+#define MICROPY_HW_FMC_BA1          (pin_G5)
+#define MICROPY_HW_FMC_NBL0         (pin_E0)
+#define MICROPY_HW_FMC_NBL1         (pin_E1)
+#define MICROPY_HW_FMC_A0           (pin_F0)
+#define MICROPY_HW_FMC_A1           (pin_F1)
+#define MICROPY_HW_FMC_A2           (pin_F2)
+#define MICROPY_HW_FMC_A3           (pin_F3)
+#define MICROPY_HW_FMC_A4           (pin_F4)
+#define MICROPY_HW_FMC_A5           (pin_F5)
+#define MICROPY_HW_FMC_A6           (pin_F12)
+#define MICROPY_HW_FMC_A7           (pin_F13)
+#define MICROPY_HW_FMC_A8           (pin_F14)
+#define MICROPY_HW_FMC_A9           (pin_F15)
+#define MICROPY_HW_FMC_A10          (pin_G0)
+#define MICROPY_HW_FMC_A11          (pin_G1)
+#define MICROPY_HW_FMC_A12          (pin_G2)
+#define MICROPY_HW_FMC_D0           (pin_D14)
+#define MICROPY_HW_FMC_D1           (pin_D15)
+#define MICROPY_HW_FMC_D2           (pin_D0)
+#define MICROPY_HW_FMC_D3           (pin_D1)
+#define MICROPY_HW_FMC_D4           (pin_E7)
+#define MICROPY_HW_FMC_D5           (pin_E8)
+#define MICROPY_HW_FMC_D6           (pin_E9)
+#define MICROPY_HW_FMC_D7           (pin_E10)
+#define MICROPY_HW_FMC_D8           (pin_E11)
+#define MICROPY_HW_FMC_D9           (pin_E12)
+#define MICROPY_HW_FMC_D10          (pin_E13)
+#define MICROPY_HW_FMC_D11          (pin_E14)
+#define MICROPY_HW_FMC_D12          (pin_E15)
+#define MICROPY_HW_FMC_D13          (pin_D8)
+#define MICROPY_HW_FMC_D14          (pin_D9)
+#define MICROPY_HW_FMC_D15          (pin_D10)
+
+#define MICROPY_HW_USB_VID                      0x2341
+#define MICROPY_HW_USB_PID                      0x0566
+#define MICROPY_HW_USB_PID_CDC_MSC              (MICROPY_HW_USB_PID)
+#define MICROPY_HW_USB_PID_CDC_HID              (MICROPY_HW_USB_PID)
+#define MICROPY_HW_USB_PID_CDC                  (MICROPY_HW_USB_PID)
+#define MICROPY_HW_USB_PID_MSC                  (MICROPY_HW_USB_PID)
+#define MICROPY_HW_USB_PID_CDC2_MSC             (MICROPY_HW_USB_PID)
+#define MICROPY_HW_USB_PID_CDC2                 (MICROPY_HW_USB_PID)
+#define MICROPY_HW_USB_PID_CDC3                 (MICROPY_HW_USB_PID)
+#define MICROPY_HW_USB_PID_CDC3_MSC             (MICROPY_HW_USB_PID)
+#define MICROPY_HW_USB_PID_CDC_MSC_HID          (MICROPY_HW_USB_PID)
+#define MICROPY_HW_USB_PID_CDC2_MSC_HID         (MICROPY_HW_USB_PID)
+#define MICROPY_HW_USB_PID_CDC3_MSC_HID         (MICROPY_HW_USB_PID)
```

Comment:
Did you test these PIDs on a Windows machine?

I'm not sure if it's still the case, but back in the early days of the PYBv1.0, Windows did not recognise the USB device if the configuration changed but the VID/PID stayed the same.  Eg if Windows first saw the USB device with a given PID as a CDC device, then that PID would forever be associated with a CDC device only.  If it then appeared as a CDC+MSC (with the same PID) then it either wouldn't work, or would only be a CDC device.

Reply:
No, but the board is assigned a single PID anyway (same for all of the other Arduino boards).

Reply:
I tested Portenta H7 on Windows 10, where it uses the same PID.  I tested switching between VCP, VCP+HID and VCP+MSC and it worked.

I guess if you don't install a driver explicitly then Windows is more flexible.

Reply:
Makes sense, thanks for testing!